jsonschema-validation.xml

```diff
@@ -503,28 +504,42 @@
             </section>
         </section>
 
-        <section title='A Vocabulary for Semantic Validation With "format"' anchor="format">
+        <section title='A Vocabulary for Semantic Content With "format"' anchor="format">
 
             <section title="Foreword">
                 <t>
-                    Structural validation alone may be insufficient to validate that an instance
-                    meets all the requirements of an application. The "format" keyword is defined to
-                    allow interoperable semantic validation for a fixed subset of values which are
+                    Structural validation alone may be insufficient to allow an application to correctly
+                    utilize certain values.  The "format" annotation keyword is defined to allow schema
+                    authors to convey semantic information for a fixed subset of values which are
                     accurately described by authoritative resources, be they RFCs or other external
                     specifications.
                 </t>
 
+                <t>
+                    Implementations MAY treat "format" as an assertion in addition to an annotation,
+                    and attempt to validate the values conformance to the specified semantics.
+                    See the Implementation Requirements below for details.
+                </t>
+
                 <t>
                     The value of this keyword is called a format attribute. It MUST be a string. A
                     format attribute can generally only validate a given set of instance types. If
                     the type of the instance to validate is not in this set, validation for this
-                    format attribute and instance SHOULD succeed.
+                    format attribute and instance SHOULD succeed.  All format attributes defined
```

Comment:
> If the type of the instance to validate is not in this set, validation for this format attribute and instance SHOULD succeed.

I've never noticed this before, but it sounds not what you want...
Say you have a format "smallint". If a string is provided at the applicable instance, validation will assert TRUE if the format says it is only applicable to numeric types.

I guess this makes sense, because it's similar to how other keywords work, in that keywords are often only applicable if they target an instance of the same type.

For example, `maxLength: 5` fails validation for a string of longer than 5 characters, but does not fail validation for a numeric value `123456789` because it is not the correct applicable type.

This is covered explicitly in draft-8 core (http://json-schema.org/work-in-progress/WIP-jsonschema-core.html#rfc.section.7.4.1) 

> 7.4.1. Assertions and Instance Primitive Types
> 
> Most assertions only constrain values within a certain primitive type. When the type of the instance is not of the type targeted by the keyword, the instance is considered to conform to the assertion.
> 
> For example, the "maxLength" keyword from the companion validation vocabulary will only restrict certain strings (that are too long) from being valid. If the instance is a number, boolean, null, array, or object, then it is valid against this assertion. 

But, I feel we should maybe re-reference this section here for clarity...

Reply:
I'd rather just take the "SHOULD succeed" language out entirely.  We used to have that phrasing everywhere and this is just left over from that.  As you note, this is typical behavior, so it's better not to call it out- calling things out should be done for atypical behavior.

Reply:
Agreed.
Is there some text about "if a keyword isn't applicable to the types it may be applied to, then it is not applied and therefore is the same as a true assertion" generally?

Take `minimum`:

> The value of "minimum" MUST be a number, representing an inclusive lower limit for a numeric instance.
> 
> If the instance is a number, then this keyword validates only if the instance is greater than or exactly equal to "minimum". 

The intent is clear, but the phrasing and wording is... not very spec like.
We don't explicitly specify what happens if the instance is not a number... 
The behaviour is, in essense, undefined (unless I've missed something).

If that's the case... maybe a spec cleanup is a candidate for draft-9.

Sorry, I've got way off topic here... but it's relevant.